layouts/_default/_markup/render-link.html

```diff
@@ -1,11 +1,14 @@
 {{ $isRedirected := findRE "^(|../|/docs/(languages|platforms)/.+?/)(api|daily-builds)(/.*)?(\\?.*)?$" .Destination -}}
 {{ $isExternal := or (hasPrefix .Destination "http") $isRedirected -}}
+{{/* Until Hugo supports hook params (https://github.com/gohugoio/hugo/issues/6670), we'll inspect .Text. */ -}}
+{{ $noExternalIcon := in .Text "hk-no-external-icon" -}}
 <a href="{{ .Destination | safeURL }}"
   {{- with .Title }} title="{{ . }}"{{ end -}}
   {{- if $isExternal }} target="_blank" rel="noopener"{{ end -}}
 >
   {{- .Text | safeHTML -}}
-  {{ if $isExternal -}}
+  {{ if and $isExternal (not $noExternalIcon) -}}
   <i class="fas fa-external-link-alt"></i>
   {{- end -}}
-</a>
\ No newline at end of file
+</a>
+{{- /* This directive ensures that all trailing whitespace is trimmed. */ -}}
```

Comment:
Adding this because it makes the hook less brittle. Otherwise it's easy to copy this file and forget to trim whitespace from the end of the file -- e.g., https://github.com/notaryproject/notaryproject.dev/pull/40. Adding a directive like this makes it obvious for all.